Library/Homebrew/extend/ENV/shared.rb

```diff
@@ -306,4 +306,10 @@ def fetch_compiler(value, source)
       end
     end
   end
+
+  def warn_if_universal_binaries_not_supported
```

Comment:
Maybe name this `check_for_compiler_universal_support` as it's throwing an exception rather than warning.